artcommon/artcommonlib/konflux/konflux_build_record.py

```diff
@@ -0,0 +1,155 @@
+import hashlib
+import json
+import random
+import typing
+import uuid
+from datetime import datetime
+from enum import Enum
+
+from google.cloud.bigquery.table import Row
+
+from artcommonlib import logutil
+
+LOGGER = logutil.get_logger(__name__)
+
+
+class KonfluxBuildOutcome(Enum):
+    FAILURE = 'failure'
+    SUCCESS = 'success'
+
+
+class ArtifactType(Enum):
+    RPM = 'rpm'
+    IMAGE = 'image'
+
+
+class Engine(Enum):
+    KONFLUX = 'konflux'
+    BREW = 'brew'
+
+
+class KonfluxBuildRecord:
```

Comment:
Not required, but maybe it is more clear to use [dataclass](https://docs.python.org/3/library/dataclasses.html) for this type?

Reply:
This class would still need custom logic to assign `nvr`, `build_id` or `record_id` which can either:
- be left blank, and auto-generate them as needed
- provided in the constructor, as it would happen when building a BuildDataRecord object from a ResultRow (BigQuery result)

I'd rather leave it as-is not to make it overly complicated